### ISSUE TITLE
[Bug]: Bus error occurs when using Rspack v0.7.2

### ISSUE THREAD

**Issue by shiqimei** [web-infra-dev/rspack#6775]
### Version

```sh
Progress: resolved 1, reused 1, downloaded 0, added 1, done

  System:
    OS: macOS 14.5
    CPU: (8) x64 Apple M1
    Memory: 203.43 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 125.0.6422.142
    Safari: 17.5
  npmPackages:
    @rsbuild/core: 0.7.4 => 0.7.4
    @rsbuild/plugin-image-compress: 0.7.4 => 0.7.4
    @rsbuild/plugin-lightningcss: 0.7.4 => 0.7.4
    @rsbuild/plugin-react: 0.7.4 => 0.7.4
    @rsbuild/plugin-sass: 0.7.4 => 0.7.4
```

### Details

> app@1.0.0 dev packages/app
> rsbuild dev

  Rsbuild v0.7.5

start   Compiling...
[1]    83988 bus error  pnpm dev
I pnpm i
@curie/react is linked to packages/app/node_modules from packages/storybook
 WARN  2 deprecated subdependencies found: glob@7.2.3, inflight@1.0.6
Packages: +11 -11
+++++++++++-----------
Downloading registry.npmjs.org/@rspack/binding-darwin-x64/0.7.1: 19.19 MB/19.19 MB, done
Progress: resolved 1100, reused 1010, downloaded 10, added 11, done

devDependencies:
- @rsbuild/core 0.7.5
+ @rsbuild/core 0.7.4 (0.7.5 is available)
- @rsbuild/plugin-image-compress 0.7.5
+ @rsbuild/plugin-image-compress 0.7.4 (0.7.5 is available)
- @rsbuild/plugin-lightningcss 0.7.5
+ @rsbuild/plugin-lightningcss 0.7.4 (0.7.5 is available)
- @rsbuild/plugin-react 0.7.5
+ @rsbuild/plugin-react 0.7.4 (0.7.5 is available)
- @rsbuild/plugin-sass 0.7.5
+ @rsbuild/plugin-sass 0.7.4 (0.7.5 is available)

Done in 13.2s
I pnpm dev

> app@1.0.0 dev packages/app
> rsbuild dev

  Rsbuild v0.7.4

start   Compiling...
ready   Client compiled in 3.19 s

### Reproduce link

n/a

### Reproduce Steps

0.7.4 is ok, please reproduce it using a arm mac.

**Comment by chenjiahan** [comment]
This might be a bug of Rspack, can you try this:

```bash
RUST_MIN_STACK=4194304 rsbuild dev
RUST_MIN_STACK=4194304 rsbuild build
```

**Comment by chenjiahan** [comment]
Rsbuild v0.7.6 has been released, and reverted `@rspack/core` to v0.7.1.

We will further investigate the root cause of the problem in `@rspack/core`.

**Comment by h-a-n-a** [comment]
If you found out this is not fixed in the next version, please reopen.
